src/Handlebars/Context.php

```diff
@@ -290,8 +297,18 @@ private function _splitVariableName($variableName)
         $bad_chars = preg_quote(self::NOT_VALID_NAME_CHARS, '/');
         $bad_seg_chars = preg_quote(self::NOT_VALID_SEGMENT_NAME_CHARS, '/');
 
-        $name_pattern = "(?:[^" . $bad_chars . "\s]+)|(?:\[[^" . $bad_seg_chars . "]+\])";
-        $check_pattern = "/^((" . $name_pattern . ")\.)*(" . $name_pattern  . ")\.?$/";
```

Comment:
Oh sorry, I actually agree with you, but CS gave me that 'too long' error. I have OCD with showing big red Xs next to my commits :(